apps/dotcom/client/src/utils/posthog.tsx

```diff
@@ -0,0 +1,101 @@
+import posthog, { PostHogConfig } from 'posthog-js'
+import 'posthog-js/dist/web-vitals'
+import { useEffect } from 'react'
+import { useLocation } from 'react-router-dom'
+import { useValue, warnOnce } from 'tldraw'
+import { useApp } from '../tla/hooks/useAppState'
+
+export type AnalyticsOptions =
+	| {
+			optedIn: true
+			user: {
+				id: string
+				name: string
+				email: string
+			}
+	  }
+	| {
+			optedIn: false
+			user?: {
+				id: string
+				name: string
+				email: string
+			}
+	  }
+
+// @ts-ignore this is fine
+const POSTHOG_KEY: string | undefined = import.meta.env.VITE_POSTHOG_KEY
+const shouldUsePosthog = POSTHOG_KEY !== undefined
+
+let currentOptions: AnalyticsOptions | null = null
+
+function configurePosthog(options: AnalyticsOptions) {
+	if (!shouldUsePosthog) return
+
+	const config: Partial<PostHogConfig> = {
+		api_host: '/api/ph',
+		capture_pageview: false,
+		persistence: options.optedIn ? 'localStorage+cookie' : 'memory',
```

Comment:
if optedIn is false why `memory` instead of just not being configured at all, i.e. just making it synonymous with `shouldUsePosthog`?